src/openvic-simulation/modifier/StaticModifierCache.cpp

```diff
@@ -1,115 +1,166 @@
+	very_hard_ai {"very_hard_ai", {}, STATIC},
+	base_modifier {"base_values", {}, STATIC},
+	war {"war", {}, STATIC},
+	peace {"peace", {}, STATIC},
+	disarming {"disarming", {}, STATIC},
+	war_exhaustion {"war_exhaustion", {}, STATIC},
+	infamy {"badboy", {}, STATIC},
+	debt_default_to {"debt_default_to", {}, STATIC},
+	great_power {"great_power", {}, STATIC},
+	secondary_power {"second_power", {}, STATIC},
+	civilised {"civ_nation", {}, STATIC},
+	uncivilised {"unciv_nation", {}, STATIC},
+	literacy {"average_literacy", {}, STATIC},
+	plurality {"plurality", {}, STATIC},
+	generalised_debt_default {"generalised_debt_default", {}, STATIC},
+	total_occupation {"total_occupation", {}, STATIC},
+	total_blockaded {"total_blockaded", {}, STATIC},
 	in_bankruptcy { nullptr },
+	bad_debtor { nullptr },
 	// Province modifiers
-	overseas { nullptr },
-	coastal { nullptr },
-	non_coastal { nullptr },
-	coastal_sea { nullptr },
-	sea_zone { nullptr },
-	land_province { nullptr },
-	blockaded { nullptr },
-	no_adjacent_controlled { nullptr },
-	core { nullptr },
-	has_siege { nullptr },
-	occupied { nullptr },
-	nationalism { nullptr },
-	infrastructure { nullptr } {}
+	overseas {"overseas", {}, STATIC},
+	coastal {"coastal", {}, STATIC},
+	non_coastal {"non_coastal", {}, STATIC},
+	coastal_sea {"coastal_sea", {}, STATIC},
+	sea_zone {"sea_zone", {}, STATIC},
+	land_province {"land_province", {}, STATIC},
+	blockaded {"blockaded", {}, STATIC},
+	no_adjacent_controlled {"no_adjacent_controlled", {}, STATIC},
+	core {"core", {}, STATIC},
+	has_siege {"has_siege", {}, STATIC},
+	occupied {"occupied", {}, STATIC},
+	nationalism {"nationalism", {}, STATIC},
+	infrastructure {"infrastructure", {}, STATIC} {}
 
-bool StaticModifierCache::load_static_modifiers(ModifierManager& modifier_manager) {
+bool StaticModifierCache::load_static_modifiers(ModifierManager& modifier_manager,ast::NodeCPtr const& root) {
 	bool ret = true;
+	key_map_t key_map {};
 
-	const auto set_static_modifier = [&modifier_manager, &ret](
-		Modifier const*& modifier, std::string_view name, fixed_point_t multiplier = 1
+	const auto set_static_country_modifier = [&key_map, &modifier_manager, &ret](
+		Modifier& modifier
 	) -> void {
-		Modifier* mutable_modifier = modifier_manager.static_modifiers.get_item_by_identifier(name);
+		ret &= add_key_map_entry(
+			key_map, modifier.get_identifier(), ONE_EXACTLY,
+			expect_dictionary(modifier_manager.expect_base_country_modifier(modifier))
+		);
+	};
 
-		if (mutable_modifier != nullptr) {
-			if (multiplier != fixed_point_t::_1()) {
-				*mutable_modifier *= multiplier;
+	const auto set_country_event_modifier = [&key_map, &modifier_manager, &ret](
+		Modifier const*& modifier_ptr, std::string_view const&& identifier, IconModifier::icon_t const& icon
+	) -> void {
+		ret &= add_key_map_entry(
+			key_map, identifier, ONE_EXACTLY,
+			[&modifier_manager, &modifier_ptr, &identifier, &icon]
+			(ast::NodeCPtr const& value)->bool {
```

Comment:
```suggestion
			[&modifier_manager, &modifier_ptr, &identifier, &icon](ast::NodeCPtr value) -> bool {
```
Formatting - should be on one line, spaces around the `->`, and ofc no `const&`